src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -609,6 +609,6 @@ extern "C" void CryptoNative_SslGet0AlpnSelected(SSL* ssl, const uint8_t** proto
 
 extern "C" int32_t CryptoNative_SslSetTlsExtHostName(SSL* ssl, const uint8_t* name)
 {
-    return static_cast<int32_t>(SSL_set_tlsext_host_name(ssl, name));
+    return static_cast<int32_t>(SSL_set_tlsext_host_name(ssl, const_cast<unsigned char*>(name)));
```

Comment:
Looking at the OpenSSL doc (https://www.openssl.org/docs/manmaster/man3/SSL_get_servername_type.html), the parameter type is `const char *` so I wonder if the error wasn't rather complaining about the cast from `const uint8_t *` to `const unsigned char*`?

Reply:
I looked at the header (not docs ;) ), and it looks like it is casting to `(char *)` instead: https://github.com/openssl/openssl/blob/773da093b1b9a63ae9c94cae756848011686caa0/ssl/tls1.h#L333-L334. My C is not great, so maybe I am missing something.

Reply:
And here is the error on my machine without this fix:

    [ 79%] Building CXX object System.Security.Cryptography.Native/CMakeFiles/objlib.dir/pal_ssl.cpp.o
    /home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp:612:63: error: cast from 'const unsigned char *' to 'char *' drops const qualifier [-Werror,-Wcast-qual]
        return static_cast<int32_t>(SSL_set_tlsext_host_name(ssl, name));
                                                                  ^
1 error generated.

Reply:
Ok, thank you for checking.